pkg/vgmanager/vgmanager_controller.go

```diff
@@ -393,11 +441,83 @@ func (r *VGReconciler) addDevicesToVG(vgName string, devices []internal.BlockDev
 	return nil
 }
 
-// filterMatchingDevices returns unmatched and matched blockdevices
-// TODO: Implement this
-func filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
-	// currently just match all devices
-	return []internal.BlockDevice{}, blockDevices, nil
+// filterMatchingDevices returns matched blockdevices
+func (r *VGReconciler) filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, error) {
+
+	var filteredBlockDevices []internal.BlockDevice
+
+	if volumeGroup.Spec.DeviceSelector != nil && len(volumeGroup.Spec.DeviceSelector.Paths) > 0 {
+		vgs, err := ListVolumeGroups(r.executor)
+		if err != nil {
+			return []internal.BlockDevice{}, fmt.Errorf("failed to list volume groups. %v", err)
+		}
+
+		for _, path := range volumeGroup.Spec.DeviceSelector.Paths {
+			diskName, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				err = fmt.Errorf("unable to find symlink for disk path %s: %v", path, err)
+				return []internal.BlockDevice{}, err
+			}
+
+			isAlreadyExist := isDeviceAlreadyPartOfVG(vgs, diskName, volumeGroup)
+			if isAlreadyExist {
+				continue
+			}
+
+			blockDevice, ok := hasExactDisk(blockDevices, diskName)
+
+			if filepath.Dir(path) == internal.DiskByPathPrefix {
```

Comment:
why can't we use this flow for all `/dev/disk/by-*` options?

Reply:
We can, We need more conditions to do that. Also as of now, all other patterns other than diskbypath will go into the else condition starting with dev.